### PR TITLE
Zephyr: Use mbed TLS allocator

### PR DESCRIPTION
**Pull request by d3zd3z**
Instead of the overly-simplistic allocator in Zephyr, use the still
simplistic allocator in mbed TLS.

On K64f, this saves 848 bytes of text, 44 bytes of data, and 208 bytes
of bss.

Signed-off-by: David Brown <david.brown@linaro.org>

### CODE REVIEW

File: boot/zephyr/os.c [hunk 1]
```diff
@@ -24,29 +24,21 @@
...
-    size_t total = nelem * size;
-    void *buf = k_malloc(total);
-    if (buf) {
-        memset(buf, 0, total);
-    }
-    return buf;
-}
+/*
+ * This is the heap for mbed TLS.  The value needed depends on the key
+ * size and algorithm used.  For RSA-2048, 6144 bytes seems to be
```
Comment (mbolivar) [inline boot/zephyr/os.c]: Looks good to me, but how was this tested? Did you run STACK_ANALYZE() in zephyr or something like that?

File: boot/zephyr/prj-p256.conf [hunk 2]
```diff
@@ -11,8 +11,8 @@ CONFIG_MBEDTLS_CFG_FILE="config-asn1.h"
 # CONFIG_TINYCRYPT_ECC_DSA is not set
 # CONFIG_TINYCRYPT_SHA256 is not set
 
-### mbedTLS wants a heap
-CONFIG_HEAP_MEM_POOL_SIZE=16384
+### mbedTLS has its own heap
+# CONFIG_HEAP_MEM_POOL_SIZE is not set
```
Comment (mbolivar) [inline boot/zephyr/prj-p256.conf]: Not familiar with this API -- is it necessary to have an "is not set" line, or can this be removed from the prj.conf completely?

Comment (d3zd3z) [reply boot/zephyr/prj-p256.conf]: I wasn't quite sure what the best way was. Leaving it out causes it to be "unchanged" from any default. The "is not set" will explicitly unset it, even if there is a default. I did this to try and make sure to remove any heap if it were to become set in Zephyr's default.

File: boot/zephyr/os.c (hunk 1 above)
Comment (d3zd3z) [reply boot/zephyr/os.c]: I couldn't get any of the heap analysis code in mbed TLS to work (it didn't even compile, and didn't work once I fixed those problems). I ended up doing kind of a binary search, and being a bit generous.

**Pull request merged**